nodejs/week3/warmup/index.js

```diff
@@ -0,0 +1,57 @@
+const { response, request } = require("express");
+const express = require("express");
+const app = express();
+const router = express.Router();
+
+app.use(express.urlencoded());
+
+router.get("/multiply", (request, response) => {
+    try {
+        const multiplyResult = Object.values(request.query)
+            .reduce((accumulator, currentValue) => accumulator.concat(currentValue), [])
+            .map((param) => Number(param))
+            .reduce((multiply, currentNum) => multiply * currentNum);
+        isNaN(multiplyResult) ? response.status(400).send("Invalid input") : response.send(`${multiplyResult}`);
+    } catch (error) {
+        response.status(404).send("values undefined");
+    }
+});
+
+router.get("/add", (request, response) => {
+    try {
+        const sumResult = Object.values(request.query)
+            .reduce((accumulator, currentValue) => accumulator.concat(currentValue), [])
+            .map((param) => Number(param))
+            .reduce((sum, currentNum) => sum + currentNum);
+        isNaN(sumResult) ? response.status(400).send("Invalid input") : response.send(`${sumResult}`);
+    } catch (error) {
+        response.status(404).send("values undefined");
+    }
+})
+
+router.post("/multiply", (request, response) => {
+    try {
+        const multiplyResult = Object.values(request.body)
+            .reduce((accumulator, currentValue) => accumulator.concat(currentValue), [])
+            .map((param) => Number(param))
+            .reduce((multiply, currentNum) => multiply * currentNum);
+        isNaN(multiplyResult) ? response.status(400).send("Invalid input") : response.send(`${multiplyResult}`);
+    } catch (error) {
+        response.status(404).send("values undefined");
+    }
+})
+
+router.post("/division", (request, response) => {
+    try {
+        const divisionResult = Object.values(request.body)
+            .reduce((accumulator, currentValue) => accumulator.concat(currentValue), [])
+            .map((param) => Number(param))
+            .reduce((devide, currentNum) => devide / currentNum);
+        isNaN(divisionResult) ? response.status(400).send("Invalid input") : response.send(`${divisionResult}`);
+    } catch (error) {
+        response.status(404).send("values undefined");
+    }
+})
+app.use("/calculator", router);
```

Comment:
Nice!

Reply:
Thanks for your time and feedback

Reply:
> Good job 💪
> 
> * You are missing get requests for subtraction and division, and post for addition and subtraction
> * Can you try to optimize your code a little bit, some parts of the code are repeating, for example `isNaN(result) ? response.status(400).send("Invalid input") : response.send(`${result}`);`
> * Please format files with Prettier https://marketplace.visualstudio.com/items?itemName=esbenp.prettier-vscode

hey @Hani-far I expected that you would do things that I suggested before the merge ☝️

Reply:
> > Good job 💪
> > 
> > * You are missing get requests for subtraction and division, and post for addition and subtraction
> > * Can you try to optimize your code a little bit, some parts of the code are repeating, for example `isNaN(result) ? response.status(400).send("Invalid input") : response.send(`${result}`);`
> > * Please format files with Prettier https://marketplace.visualstudio.com/items?itemName=esbenp.prettier-vscode
> 
> hey @Hani-far I expected that you would do things that I suggested before the merge ☝️

hey, @cecastosic Yes, Sure I am doing that, I had 2 open repos and the problem is I merged this one by mistake instead of the other one. I didn't know what to do, I was thinking to make changes and push them again. Thanks again for the suggestions.

Reply:
@cecastosic ,I made changes and created a new PR, make I ask you to review it when you have time?